src/ngraph_deassign_clusters.cc

```diff
@@ -170,6 +215,8 @@ Status DeassignClusters(Graph* graph) {
         node->ClearAttr("_ngraph_cluster");
         // TODO(amprocte): move attr name to a constant
         node->ClearAttr("_ngraph_marked_for_clustering");
+
+        deassigned_histogram[node->type_string()]++;
```

Comment:
In our discussion we identified 4 tiers of ops:
1) Rejected in all forms
2) Rejected under certain static conditions:
     2a. confirmationfunction passes
     2b. datatypes match
3) Rejected under some dynamic conditions (resulting from autoclustering) (too small cluster)
4) Accepted

We had decided to print out histograms of (1), (2) and (3).

Below is a list of actionable items from these histograms:
*Histogram of (1)*: This tells us that an op appears a lot in the graph, but we don't support it, so we probably should think of supporting it
*Histogram of (2)*: For an op is "partially" supported, this histogram will remind us that the op is rejected because of the cases we do not cover. For example we only do "scaled" mode in `QuantizeV2` op. If `QuantieV2` starts showing up in this histogram, we will be made aware that we are seeing graphs where our bridge is failing to accept these and we can then decide to cover those gaps
*Histogram of (3)*: A histogram of ops that were declustered. Note that category (1) and (2) are static categories. Category (3) is dynamic, in the sense results can't be predicted till we run the auto-cluster algorithm. Then it is not really the "fault" of the node that it landed up in a cluster of 1 node only and subsequently got declustered. That is knowing that "const" got declustered 5 times and "add" got declustered 10 times is not telling us much (or is it?). Lets discuss tomorrow with @avijit-nervana what this histogram represents and how it can be turned into an actionable item.


Of course the other correction you made in the PR is still valid (relating to `num_nodes_marked_before_deassign`) which now gets rid of the `100%` bug

Reply:
Thanks for the detailed explanation!

Regarding the third histogram with the declustered node, it may help us:
1. Make sure important nodes are not declustered. If they are, probably we want to look at the graph to analyse it more. 
2. The information may help us when we do model analysis on the bridge. Then we know which part of the graph split to look at, and maybe cross checking with your edge histogram to make sure there is no other case that prevents the node being clustered with others.

Those are my thoughts. We can discuss more today!